Added fallback to default credential provider chain.

christianblunden opened the pull request:
It is probably best to use the DefaultCredentialProviderChain for authentication as it allows use of environment variables, IAM roles and credential files.  I would take out the BasicAWSCredentials provider completely but have left it in for back compatibility.
see http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/credentials.html

Had to remove the licence file mapping as it is no longer present in the latest SDK

Also increased the version of clojure to 1.6.0 and aws sdk to 1.8.9.1.

I increased the version of the jar to 0.5.0 since I haven't tested that the latest SDK works for all methods.

christianblunden committed:
Added fallback to default credential provider chain.

buildhive commented:
[Michael Rowe » clj-aws-ec2 #58](https://buildhive.cloudbees.com/job/mrowe/job/clj-aws-ec2/58/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

mrowe commented:
Hi Christian.

Thanks for that!

Your patch looks great, except I'd really rather keep the Clojure version as low as possible so that the library is accessible to as many people as possible. Can you roll another patch without that bit?

Cheers,
Michael Rowe

buildhive commented:
[Michael Rowe » clj-aws-ec2 #59](https://buildhive.cloudbees.com/job/mrowe/job/clj-aws-ec2/59/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Michael Rowe » clj-aws-ec2 #60](https://buildhive.cloudbees.com/job/mrowe/job/clj-aws-ec2/60/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

christianblunden commented:
Hey no problem :) didn't consider the clojure version just increased it out of habit.
I've dropped it back to 1.2.1 and rebased the original pull request. 

cheers

mrowe commented:
Thanks Christian. Look, sorry to be a pain, but can you split the SDK version bump into its own commit? And also the project version change (or leave that out completely and I'll do it when I cut a new release). Small, focused commits make it easier to see what's going on (and revert if necessary).

Thanks again for your contribution, I really appreciate it!

buildhive commented:
[Michael Rowe » clj-aws-ec2 #61](https://buildhive.cloudbees.com/job/mrowe/job/clj-aws-ec2/61/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

christianblunden committed:
raise version of SDK to 1.8.9.1.

buildhive commented:
[Michael Rowe » clj-aws-ec2 #62](https://buildhive.cloudbees.com/job/mrowe/job/clj-aws-ec2/62/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

christianblunden commented:
Hey no problem. I've split the change into 2 commits. The first only adds the Default Credential Provider and nothing else and the second increases the SDK version to latest 1.8.9.1.  I've left the jar version change out.. 

It should also be noted that between v1.6.1 and v1.8.9.1 of the SDK the ENV variables required to be set change slightly.

v1.6.1
AWS_ACCESS_KEY_ID
AWS_SECRET_KEY

https://github.com/aws/aws-sdk-java/blob/1.6.1/src/main/java/com/amazonaws/auth/EnvironmentVariableCredentialsProvider.java

v1.8.9.1
AWS_ACCESS_KEY_ID
AWS_SECRET_ACCESS_KEY

https://github.com/aws/aws-sdk-java/blob/1.8.9.1/src/main/java/com/amazonaws/SDKGlobalConfiguration.java

cheers

mrowe commented:
Awesome, thanks!

I'll get out a release as soon as I get a chance.
